### PR TITLE
Select metadata headers cleanup

### LINKED ISSUE #781: Provide basic instructions in Metadata Selector
OPUS 1 (RIP) included some information on what to actually do in the Metadata Selector ("Drag from here" and "Drop here"). It's not a bad idea to do something like that now. I would suggest putting a header on the left pane "Click to select or deselect metadata" and the right pane "Selected metadata\n(Used for table view, slideshow, and CSV files).

### PR DESCRIPTION
**Pull request by rfrenchseti**
- Fixes #781 
- Clean up select metadata code
- Rename metadata selector to select metadata everywhere
- Add headers to select metadata dialog with instructions
- Implement dialog responsiveness in JS instead of CSS for more flexibility
- Rename #searchMenu to op-search-menu and various other cleanup

### CODE REVIEW

File: opus/application/static_media/js/browse.js [hunk 1]
```diff
@@ -1325,20 +1294,170 @@ var o_browse = {
...
+           So we use a default value that will cover the common case of a dialog wide
+           enough to cause excessive header word wrap.
+        */
+        if (modalHeaderHeight === 0) {
+            modalHeaderHeight = 57;
+            modalFooterHeight = 68;
+            selectMetadataHeadersHeight = 122;
+            selectMetadataHeadersHRHeight = 1;
+
+        }
```
Comment (juzen2003) [inline opus/application/static_media/js/browse.js:1394]: We can remove this extra line above in line 1394.

File: opus/application/static_media/js/browse.js [hunk 2]
```diff
@@ -1325,20 +1294,170 @@ var o_browse = {
...
+        let totalNonGalleryHeight = (footerHeight + mainNavHeight + modalHeaderHeight +
+                                     modalFooterHeight + selectMetadataHeadersHeight +
+                                     selectMetadataHeadersHRHeight);
```
Comment (juzen2003) [inline opus/application/static_media/js/browse.js]: Do we want to rename totalNonGalleryHeight? This is the height of non-scrollable area (height of everything other than .op-no-select) when select metadata modal is open.

File: opus/application/static_media/js/opus.js [hunk 3]
```diff
@@ -617,8 +617,9 @@ var opus = {
         let adjustProductInfoHeightDB = _.debounce(o_cart.adjustProductInfoHeight, 200);
         let adjustDetailHeightDB = _.debounce(o_detail.adjustDetailHeight, 200);
         let adjustHelpPanelHeightDB = _.debounce(opus.adjustHelpPanelHeight, 200);
-        let adjustMetadataSelectorMenuPSDB = _.debounce(o_browse.adjustMetadataSelectorMenuPS, 200);
-        let adjustSelectedMetadataPSDB = _.debounce(o_browse.adjustSelectedMetadataPS, 200);
+        let adjustSelectMetadataHeightDB = _.debounce(o_browse.adjustSelectMetadataHeight, 200);
```
Comment (juzen2003) [inline opus/application/static_media/js/opus.js]: Just a note: we can use non-debounced version ```let adjustSelectMetadataHeightDB = o_browse.adjustSelectMetadataHeight;``` to remove the jumping when browser width is changing around 1000px & 1001px, 775px & 774px, and 652px & 653px.

**Review by juzen2003 — CHANGES_REQUESTED** [review]
- Just two minor changes needed.
- Responsiveness works nicely with adjustSelectMetadataHeight. A note of observation, we will see a little height jumping when text in .op-select-metadata-headers are wrapping up to a new line or shrinking to reduce one line (when browser width is changing around 1000px & 1001px, 775px & 774px, and 652px & 653px). To remove that jumping while adjustSelectMetadataHeight is running the calculation, we can use the non-debounced version in "resize" event handler (line 620 in opus.js).

File: opus/application/static_media/js/browse.js (hunk 1 above)
Comment (rfrenchseti) [reply opus/application/static_media/js/browse.js:1394]: Done

File: opus/application/static_media/js/browse.js (hunk 2 above)
Comment (rfrenchseti) [reply opus/application/static_media/js/browse.js]: Rename to totalNonScrollableHeight

File: opus/application/static_media/js/opus.js (hunk 3 above)
Comment (rfrenchseti) [reply opus/application/static_media/js/opus.js]: Changed

**Pull request merged**